src/python_routing_v02/compute_nhd_routing_SingleSeg.py

```diff
@@ -497,15 +507,21 @@ def main():
             print("executing computation on ordered reaches ...")
 
         #########
-        waterbodies_values = supernetwork_values[13]
+        waterbodies_values = supernetwork_values[12]
+        waterbodies_segments = supernetwork_values[13]
         
         for terminal_segment, network in networks.items():
-            is_reservoir = terminal_segment in waterbodies_values
+            # is_reservoir = terminal_segment in waterbodies_segments
+            try:
+                waterbody = waterbodies_segments[terminal_segment]
+            except:
```

Comment:
Maybe slightly faster as 
```
waterbody = None
try:
    waterbody = waterbodies_segments[terminal_segment]
except:
    pass
```

Reply:
Actually, you want `waterbody.get(terminal_segment)`

Reply:
If you need to handle the exception, then go the `try/except` route, but if you just need the key or some default/sentinel value, `dict.get` is the way to go.